rs/replicated_state/src/canister_state/system_state.rs

```diff
@@ -268,6 +271,219 @@ impl CanisterHistory {
     }
 }
 
+/// `TaskQueue` represents the implementation of a queue structure for tasks.
```

Comment:
I'd expand a bit on this doc comment. What I think would be useful to list (for someone to get an overview):

1. There are hooks (now one but in the future more) that if `Ready`, need to be executed before anything else.
2. Other tasks are executed in order from a FIFO queue.
3. List any invariants that the data structure enforces.

Reply:
That was written previously:
```
/// `TaskQueue` represents the implementation of queue structure for tasks satisfying  conditions:
///
/// 1. If there is a `Paused` or `Aborted` task it will be returned first.
/// 2. If an `OnLowWasmMemoryHook` is ready to be executed, it will be returned next.
/// 3. All other tasks will be returned based on the order in which they are added to the queue.
```
But it is moved to the follow-up since the `TaskQueue` does not yet implement functionality for poping hooks from the queue but only has the same behaviour as before this PR.